src/macros.rs

```diff
@@ -189,12 +189,15 @@ pub fn add_child<MSG>(stack: &mut Stack<MSG>, child: VNode<MSG>) {
 #[doc(hidden)]
 pub fn child_to_parent<MSG>(stack: &mut Stack<MSG>, endtag: Option<&'static str>) {
     if let Some(node) = stack.pop() {
-        let starttag = node.tag();
-        if let Some(endtag) = endtag {
-            if starttag != endtag {
-                panic!("wrong closing tag: <{}> -> </{}>", starttag, endtag);
+        {
+            let starttag = node.tag();
```

Comment:
Could you move this after `if let Some(endtag) = endtag` check? It will help you to remove unnecessary scope braces in 192 and 199 lines.